exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -607,7 +607,14 @@ public long getRowCount() {
       fileSet = Sets.newHashSet(fileNames);
     }
 
-    if (fileNames.isEmpty()) {
+    List<String> finalFileNames;
+    if (fileSet != null) {
+      finalFileNames = Lists.newArrayList(fileSet);
```

Comment:
Can we use fileNames only? From Line 589 / 607, seems fileSet is assigned from fileNames; seems they are same under two ELSE branches.